libvast/src/system/passive_partition.cpp

```diff
@@ -348,8 +348,10 @@ partition_actor::behavior_type passive_partition(
       }
       auto start = std::chrono::steady_clock::now();
       auto triples = detail::evaluate(self->state, query.expr);
-      if (triples.empty())
-        return uint64_t{0};
+      if (triples.empty()) {
+        rp.deliver(0);
+        return rp;
+      }
```

Comment:
Yes this looks sus. But I would really like to have a test for it to be sure.

Reply:
CAF 0.18 actually asserts in the destructor of a response promise that it was delivered exactly once, and the semantic of that didn't change. We were likely just in UB land here.

Reply:
Oh and you'll need to call deliver with `deliver(uint64_t{0})`.

Reply:
Shouldn't be necessary since it's a typed response promise, no?

Reply:
I added a test and verified that it times out without this change.